recipes/libssh2/all/conanfile.py

```diff
@@ -87,7 +87,7 @@ def configure(self):
         self.settings.rm_safe("compiler.cppstd")
 
     def requirements(self):
-        if self.options.with_zlib:
+        if self.options.with_zlib or self.options.crypto_backend == "openssl":
             self.requires("zlib/[>=1.2.11 <2]")
```

Comment:
zlib is already a transitive requirement of openssl3 - not sure this should be explicit on openssl alone

the find_package call was probably added to support static builds (it needs to be aware of linking transitive static dependencies), but also to support generating the correct .pc file for static dependencies - this is already handled by Conan and the .pc files are removed - so it would be unneeded/redundant
https://github.com/libssh2/libssh2/commit/67ac735ad09e48bb5c92e765c72a6564d5873c27